omas/omas_physics.py

```diff
@@ -227,15 +291,27 @@ def map_flux_coordinate_to_pol_flux(ods, time_index, origin, values):
 
     :return: Transformed values
     """
-    if origin == "rho_pol":
-        return (
-            values**2
-            * (
-                ods["equilibrium"]["time_slice"][time_index]["global_quantities"]["psi_sep"]
-                - ods["equilibrium"]["time_slice"][time_index]["global_quantities"]["psi_axis"]
-            )
-            + ods["equilibrium"]["time_slice"][time_index]["global_quantities"]["psi_axis"]
-        )
+    if origin == "rho_pol_norm":
+        return (values**2 * (ods["equilibrium"]["time_slice"][time_index]["global_quantities"]["psi_boundary"]
+                - ods["equilibrium"]["time_slice"][time_index]["global_quantities"]["psi_axis"])
+                + ods["equilibrium"]["time_slice"][time_index]["global_quantities"]["psi_axis"])
+    elif origin == "rho_tor_norm":
+        phi = values**2
+        phi *= np.array([ods["equilibrium"]["time_slice"][time_index]["global_quantities"]["psi_boundary"]]) / np.max(phi)
+        return map_flux_coordinate_to_pol_flux(ods, time_index, "phi", phi)
+    elif origin == "phi":
+        from scipy.interpolate import InterpolatedUnivariateSpline
+        psi_grid = ods["equilibrium"]["time_slice"][time_index]["profiles_1d"]["psi"]
+        psi_mask = mask_SOL(ods, time_index, psi_grid)
+        phi_grid = ods["equilibrium"]["time_slice"][time_index]["profiles_1d"]["psi"][psi_mask]
+        psi_spl = InterpolatedUnivariateSpline(phi_grid, psi_grid[psi_mask])
+        phi_min = np.min(phi_grid)
+        phi_max = np.max(phi_grid)
+        values_mask = np.logical_ant(values > phi_min, values < phi_max)
```

Comment:
np.logical_ant(values > phi_min, values < phi_max) is the same as (values > phi_min)&(values < phi_max)

Reply:
I I literally put a bug in this line xD